### PR TITLE
fix(Scripts/Arcatraz): fix Trial of the Naaru: Tenacity complete condition

### PR DESCRIPTION
**Pull request by elthehablo**
should now only complete on heroic mode

## Changes Proposed:
-  
-  

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/15728
- Closes https://github.com/chromiecraft/chromiecraft/issues/5293

## SOURCE:

the quests are all heroic only :)

## Tests Performed:

- simple test by myself done following the little guide below. it doesn't complete on final boss kill on normal, but it does on heroic! so it is working as intended!
- other people need to test as well :)

## How to Test the Changes:

1. .quest add 10886
2. go to arca normal
3. do event with last boss and millhouse
4. see (hopefully) that quest doesn't complete
5. go out and set difficulty to heroic
6. do the same event
7. see that event completes now

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/scripts/Outland/TempestKeep/arcatraz/arcatraz.cpp [hunk 1]
```diff
@@ -345,7 +345,9 @@ class npc_warden_mellichar : public CreatureScript
                 instance->SetBossState(DATA_WARDEN_MELLICHAR, DONE);
                 if (Creature* creature = summons.GetCreatureWithEntry(NPC_MILLHOUSE))
                 {
-                    instance->DoCastSpellOnPlayers(SPELL_QID10886);
+                    if (IsHeroic()){
```
Comment (heyitsbench) [inline src/server/scripts/Outland/TempestKeep/arcatraz/arcatraz.cpp:348]: [c](https://www.azerothcore.org/wiki/cpp-code-standards#brackets)

**Comment by Gultask** [comment]
Remember to add the original CC issue as well

File: src/server/scripts/Outland/TempestKeep/arcatraz/arcatraz.cpp (hunk 1 above)
Comment (elthehablo) [reply src/server/scripts/Outland/TempestKeep/arcatraz/arcatraz.cpp:348]: okay bench I will do a new commit after testing

**Comment by heyitsbench** [comment]
Tested. Quest was credited on heroic but not on normal. Works as described. 👍

**Review by Nyeriah — APPROVED** [review]

**Pull request merged**